algoliasearch-common/src/main/java/com/algolia/search/APIClient.java

```diff
@@ -968,6 +968,20 @@ void deleteByQuery(String indexName, Query query, int batchSize, RequestOptions
     }
   }
 
+  Task deleteBy(String indexName, Query query, RequestOptions requestOptions) throws AlgoliaException {
+    query = query == null ? new Query() : query;
+    AlgoliaRequest<Task> algoliaRequest = new AlgoliaRequest<>(
+      HttpMethod.POST,
+      false,
+      Arrays.asList("1", "indexes", indexName, "deleteByQuery"),
+      requestOptions,
+      Task.class
+    )
+      .setData(query);
```

Comment:
What will `setData` do? As per the [REST API Reference](https://www.algolia.com/doc/rest-api/search/#delete-by-query), the API expects a JSON object with a `params` attribute containing all parameters as a URL-encoded string.